tools/autograd/gen_autograd.py

```diff
@@ -67,6 +65,20 @@
 # returns a view of its inputs
 RETURNS_VIEWS_OF_INPUT = set(VIEW_FUNCTIONS.keys()).union({'chunk', 'split'})
 
+# note: For functions that return multiple views, we make a distinction between the
+# functions that only perform viewing (called pure views here) versus all others (like
+# user-defined Functions). This list is used to know which functions are pure views.
+# A function should NOT be added to this list if:
+# - Your backward function does more than just the backward of the views. For example if your
+# function makes changes to the layout both in forward and backward.
+# - You want to use hooks with that function
+# A function should be added to this list if:
+# - You want to allow inplace modification of the result.
+# - You are aware that inplace modification of the output will change the memory usage of the
+#   backward as the multiple Nodes we create will force the gradient buffer to be created
+#   for each Node, instead of once in the original multi-output Node.
+# If you update this list, please update "test_multi_view_methods" in test_autograd.py
+PURE_VIEW_FUNCTIONS = ['split', 'split_with_sizes']
```

Comment:
nit: make it a set?

Explain to me again why the default is that all multi-return view functions are non-pure, as opposed to pure? (The opposite seems to intuitively make more sense to me)

Reply:
I was also expecting this to say something about unbind here (which morally should be in the list, but iirc you decided that you had to keep the old behavior first, right?) It's totally non-obvious as written here that the omission of unbind is intentional

Reply:
I will rework the comment to make it clearer that we only introduce this to ensure backward compatibility with functions that have a wrong behavior. But that we cannot change.

Reply:
Actually, because only the other commit introduces these functions as view, I'll just remove this from this PR.